apps/crl.c

```diff
@@ -195,6 +201,11 @@ int crl_main(int argc, char **argv)
         case OPT_MD:
             digestname = opt_unknown();
             break;
+        case OPT_ENGINE:
+#ifndef OPENSSL_NO_ENGINE
+            e = setup_engine(opt_arg(), 0);
+#endif
```

Comment:
I see this call without #ifdef in other apps...

Reply:
`setup_engine()` will get your engine registered.  Then when you go to use algorithms your pmeth's and ameth's in your engine can  be selected.  You might think this looks strange, but if you grep for `engine` in the follow apps code files you'll see my engine code is similar to them. 

- x509.c
- srp.c
- verify.c
- pkey.c 
- rsautl.c
....and many others.

Reply:
> I see this call without #ifdef in other apps...

I was just following what `OPENSSL_NO_MD5` ....if I follow that, it looks nicer and has less cognitive load.